### PR TITLE
[BAHIR-90] Include Apache Flink extensions to website setup

### PR DESCRIPTION
**Pull request by tzulitai**
This commit setups links for documentation and downloads for Apache
Bahir Flink extensions. The Flink docs are generated using the updated `update-docs.sh` script in #8.

It also revamps the website's project infrastructure, configuration, and navigation / layouts for documentations and the download page to be more friendly for newly supported platforms in the future.

Notes about change to the website's description in the title:
I tried adding "such as Apache Spark _and Apache Flink_" at the end, but the line became too long and looked a bit weird. Therefore, I ended up with just changing to "Apache Bahir provides extensions to multiple distributed analytic platforms." I'm generally open to discuss the changes to website content wordings in this PR!

Also, the links for Flink in the new download page are obviously invalid at the moment. They'll become valid as soon as Apache Bahir for Flink 1.0 is released, and we replace the `TBA` placeholder values for `flink_latest_release` and `flink_latest_release_date`  in `project.yml`.

### CODE REVIEW

**Comment by ckadner** [comment]
@tzulitai -- thanks for your PR. This looks good to me overall. Cleaner than before :-)

@lresende, @rmetzger -- what are your thoughts?

File: site/_config.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 title: Apache Bahir
 email: dev@bahir.apache.org
 description: > # this means to ignore newlines until "baseurl:"
-  Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark.
+  Apache Bahir provides extensions to multiple distributed analytic platforms.
```
Comment (ckadner) [inline site/_config.yml]: I would keep mentioning Spark and Flink ... think Google search results

"Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark and Apache Flink"

File: site/_data/project.yml [hunk 2]
```diff
@@ -15,15 +15,14 @@
...
 #
+
+# settings common across all supported platforms
+
 name: Apache Bahir
 short_name: Bahir
-unix_name: bahir
-github_project_name: bahir
-description: Apache Bahir provides extensions to distributed analytic platforms such as Apache Spark.
+description: Apache Bahir provides extensions to multiple distributed analytic platforms.
```
Comment (ckadner) [inline site/_data/project.yml]: I would keep mentioning Spark and Flink ... see above

File: site/_includes/themes/apache-clean/_navigation.html [hunk 3]
```diff
@@ -5,8 +5,8 @@
     <!--h1>{{ site.data.project.name }}</h1-->
     <p>{{ site.data.project.description }}</p>
 
-    {% if site.data.project.latest_release %}
-    <a href="{{ site.data.project.download }}" class="btn btn-primary btn-lg bigFingerButton" role="button">Download {{ site.data.project.name }} {{site.data.project.latest_release}}</a></p>
+    {% if site.data.project.download %}
+      <a href="{{ site.data.project.download }}" class="btn btn-primary btn-lg bigFingerButton" role="button">Download {{ site.data.project.name }} Latest Releases</a></p>
```
Comment (ckadner) [inline site/_includes/themes/apache-clean/_navigation.html]: I think we should add two buttons, one for Bahir Spark and one for Bahir Flink (once available) and keep the version in the button text(s)
and the test should be for `spark_latest_release` and `flink_latest_release` (barring "TBA", of course)

File: site/_includes/themes/apache/_navigation.html [hunk 4]
```diff
@@ -49,7 +49,7 @@
     <h1>{{ site.data.project.name }}</h1>
     <p>{{ site.data.project.description }}</p>
 
-    {% if site.data.project.latest_release %}
+    {% if site.data.project.download %}
```
Comment (ckadner) [inline site/_includes/themes/apache/_navigation.html]: I think we should add two buttons, one for Bahir Spark and one for Bahir Flink (once available) and keep the version in the button text(s)
and the test should be for `spark_latest_release` and `flink_latest_release` (barring "TBA", of course)

File: site/_config.yml [hunk 5]
```diff
@@ -16,7 +16,7 @@
...
 baseurl: "" # the subpath of your site, e.g. /blog
 url: "http://bahir.apache.org" # the base hostname & protocol for your site
 #twitter_username: ApacheBahir
```
Comment (ckadner) [inline site/_config.yml]: on [line 32](https://github.com/apache/bahir-website/pull/9/files#diff-0a55ba953b4a00424ce1150c8cb5e4efR32), this: `exclude: [README.md,Gemfile*,docs/spark/templates]`
should now be: `exclude: [README.md,Gemfile*,docs/*/templates]` to also exclude flink templates from generation

File: site/download.md [hunk 6]
```diff
@@ -69,37 +70,87 @@ Our latest release is {{ site.data.project.name }} {{site.data.project.latest_re
...
+
+### Previous Releases
+
+All previous releases of {{ site.data.project.name }} Spark Extensions can be found in the [archives](http://archive.apache.org/dist/{{site.data.project.spark_unix_name}}/).
+
+## Bahir Flink Extensions Latest Release 
+
+Our latest {{ site.data.project.name }} release for Apache Flink extensions is {{site.data.project.flink_latest_release}}, released on {{site.data.project.flink_latest_release_date}}.
+
+{% if site.data.project.flink_latest_release %}
```
Comment (ckadner) [inline site/download.md]: don't render this table if release is not published ("TBA")

File: site/_data/project.yml [hunk 7]
```diff
@@ -48,10 +47,31 @@ commits_list_archive_markmail:
...
+
+spark_source_repository: https://git-wip-us.apache.org/repos/asf/bahir.git
+spark_source_repository_mirror: https://github.com/apache/bahir
+
+# Apache Flink
+
+flink_unix_name: bahir-flink
+flink_github_project_name: bahir-flink
+
+flink_latest_release: TBA
```
Comment (ckadner) [inline site/_data/project.yml]: leave blank (null) to not render dependent content, like release notes, download table, download big finger button, etc

**Review by ckadner — CHANGES_REQUESTED** [review]
@tzulitai -- I rendered the Bahir website with your changes and found a few small issues to be addressed. 

> I tried adding "such as Apache Spark and Apache Flink" at the end, but the line became too long and looked a bit weird. Therefore, I ended up with just changing to "Apache Bahir provides extensions to multiple distributed analytic platforms."

Since it's only two projects so far, I would keep mentioning _Apache Spark and Apache Flink_

> Also, the links for Flink in the new download page are obviously invalid at the moment. They'll become valid as soon as Apache Bahir for Flink 1.0 is released, and we replace the `TBA` placeholder values for `flink_latest_release` and `flink_latest_release_date` in `project.yml`.

Since some content is rendered based on the version place holders, please keep the value empty (`null`) or test for the value `TBA` accordingly

Again, thanks your PR!

**Comment by tzulitai** [comment]
Thanks for the comments @ckadner. They make sense! I'll update the PR accordingly.

**Comment by tzulitai** [comment]
@ckadner -- I've hopefully addressed your comments. Please let me know if there's anything else you think we should change!

Note: Instead of allowing `TBA`, I decided to just keep it simple by checking if the `xx_latest_release` config exists.

**Comment by lresende** [comment]
Overall LGTM, just the caveat that I haven't actually tested in a staging site.

**Comment by rmetzger** [comment]
Thank you for merging.

**Pull request closed without merging**